### DISCUSSION TITLE
Next js <Image> tag with url at build time Fails to load the image

### DISCUSSION THREAD

**Discussion by lokesh14v** [vercel/next.js#20531] [Help]
NextJs <Image> tag with CDN URL gives 404 after the production build is taken

"next": "^10.0.3",

<img width="1142" alt="Screenshot 2020-12-28 at 1 10 19 PM" src="https://user-images.githubusercontent.com/26504161/103216581-4f13ce80-490e-11eb-9162-628e7f0c0867.png">

**Comment by agcty** [comment]
I'm having the same issue! But only for some images for whatever reason. This is a preview of my site where this is happening: https://gogl-io-git-feature-2.agcty.vercel.app/

**Reply by lokesh14v** [reply]
but I don't see any 404 images in your project .

**Reply by agcty** [reply]
ah sorry, wrong link

Here if you scroll down

 https://gogl-io-git-feature-2.agcty.vercel.app/blog/what-i-learned-about-startu

**Comment by ielmar** [comment]
Do you have something like below code in your next.config.js file as indicated in https://nextjs.org/docs/basic-features/image-optimization link? 

`images: {
    loader: 'imgix',
    path: 'https://s1.rdbuz.in'
  },`

After inserting this code in next.config.js file, the images on my site started to show directly, without the "_next/image/?url=...".

**Reply by agcty** [reply]
Ah I'm sorry, the issue was that I forgot to set an environment variable in vercel 😅

**Reply by lokesh14v** [reply]
> Do you have something like below code in your next.config.js file as indicated in https://nextjs.org/docs/basic-features/image-optimization link?
> 
> `images: { loader: 'imgix', path: 'https://s1.rdbuz.in' },`
> 
> After inserting this code in next.config.js file, the images on my site started to show directly, without the "_next/image/?url=...".

I gave it as domain   -domains: ['s1.rdbuz.com'],
and what exactly the purpose of **loader: 'imgix',**

**Reply by ielmar** [reply]
I have it in my site as both path and domain:
`images: {
  loader: 'imgix', path: 'https://s1.rdbuz.in', domains: ['s1.rdbuz.in']
}`

The documentation says for custom CDNs you can use any of the CDN loader but change the path to yours. It then generates absolute URL for your images with the given path.

https://nextjs.org/docs/basic-features/image-optimization#loader in this section it explains.

**Reply by lokesh14v** [reply]
> I have it in my site as both path and domain:
> `images: { loader: 'imgix', path: 'https://s1.rdbuz.in', domains: ['s1.rdbuz.in'] }`
> 
> The documentation says for custom CDNs you can use any of the CDN loader but change the path to yours. It then generates absolute URL for your images with the given path.
> 
> https://nextjs.org/docs/basic-features/image-optimization#loader in this section it explains.

Thanks, @ielmar for clarifying the doubt.
